### ISSUE TITLE
'SpecReporter' is not assignable to type 'Reporter | CustomReporter'

### ISSUE THREAD

**Issue by squidfunk** [bcaudan/jasmine-spec-reporter#588]
# Issue guidelines

## Bug report

Check with "x":

- [x] I have this issue with the latest release of [jasmine-spec-reporter](https://github.com/bcaudan/jasmine-spec-reporter/releases) 
- [x] I have this issue with the latest release of [jasmine](https://github.com/jasmine/jasmine/releases) 
- [ ] I can reproduce this issue with one of the [examples](https://github.com/bcaudan/jasmine-spec-reporter/tree/master/examples)

**Current behavior**

I'm encountering a type error when configuring the spec reporter, probably related to the latest `@types/jasmine`package:

```
index.ts(91,27): error TS2345: Argument of type 'SpecReporter' is not assignable to parameter of type 'Reporter | CustomReporter'.
  Type 'SpecReporter' is not assignable to type 'CustomReporter'.
    Types of property 'suiteStarted' are incompatible.
      Type '(result: import("node_modules/jasmine-spec-reporter/built/spec-reporter").CustomReporterResult) => void' is not assignable to type '(result: jasmine.CustomReporterResult) => void'.
        Types of parameters 'result' and 'result' are incompatible.
          Type 'jasmine.CustomReporterResult' is not assignable to type 'import("node_modules/jasmine-spec-reporter/built/spec-reporter").CustomReporterResult'.
            Types of property 'duration' are incompatible.
              Type 'number | null' is not assignable to type 'string | undefined'.
                Type 'null' is not assignable to type 'string | undefined'.
```

**Expected behavior**

No type error, assignment works as expected.

**Minimal reproduction of the problem with instructions**

Package versions:

- `@types/jasmine`: 3.6.6
- `jasmine`: 3.6.4
- `jasmine-spec-reporter`: 6.0.0

``` ts
const jasmine = new Jasmine({})
jasmine.env.clearReporters()
jasmine.env.addReporter(new SpecReporter({
  spec: { displayStacktrace: StacktraceOption.RAW }
}))
```

**Possible fix**

Change the typing of `duration` to `number | null` here:

https://github.com/bcaudan/jasmine-spec-reporter/blob/8d7bd5f78a640fc2c04dc6397c47ca18b806ac97/src/spec-reporter.ts#L20-L22

**Comment by SetupCoding** [comment]
As a workaround you can do 
`jasmine.getEnv().addReporter((new SpecReporter() as unknown) as CustomReporter);`

**Comment by squidfunk** [comment]
Thanks! I just added a `@ts-ignore` above the line with the same result. This issue was meant as an FYI that the typings might need to be updated 😊

**Comment by bcaudan** [comment]
Indeed, it seems to have been detected by dependabot as well #585 

I'll look into it, thanks for the heads up

**Comment by coyoteecd** [comment]
@bcaudan any update? I forked and looked at it, but it seems there's quite some changes involved

**Comment by bcaudan** [comment]
I have started to look into it but I have to look deeper into the spec duration from jasmine to see if there is still a need for jasmine-spec-reporter to compute it as well.
New lockdown starting this weekend for me, so I should come back to it soon 🤞

**Comment by coyoteecd** [comment]
Covid-19 helping open source development 😁

**Comment by bcaudan** [comment]
A fix is available in jasmine-spec-reporter@[7.0.0](https://github.com/bcaudan/jasmine-spec-reporter/releases/tag/v7.0.0)

**Comment by dominionikpeawujor** [comment]
> As a workaround you can do
> 
> ```ts
> import CustomReporter = jasmine.CustomReporter;
> 
> 
> jasmine.getEnv().addReporter((new SpecReporter() as unknown) as CustomReporter);
> ```

This saved me after days of debugging. Thank you @SetupCoding
